src/ProxyManager/Factory/AbstractBaseFactory.php

```diff
@@ -65,8 +66,27 @@ protected function generateProxy(string $className, array $proxyOptions = []) :
             ->configuration
             ->getClassNameInflector()
             ->getProxyClassName($className, $proxyParameters);
-
-        if (! class_exists($proxyClassName)) {
+        
+        $strategy = $this->configuration->getGeneratorStrategy();                                                  
+                                                                                                                   
+        if ($strategy instanceof FileWriterGeneratorStrategy) {                    
```

Comment:
This couples the factory implementation to an implementation of the `GeneratorStrategy` - please don't do this. The only assumption is that you get a generator strategy, but you can't upcast and check it.

Reply:
@Ocramius. Please check, it is updated.